BlueprintUI/Sources/Internal/CacheTree.swift

```diff
@@ -22,12 +22,23 @@ protocol CacheTree: AnyObject {
 extension CacheTree {
     /// Convenience method to get a cached size, or compute and store one if it is not in the cache.
     func get(_ constraint: SizeConstraint, orStore calculation: (SizeConstraint) -> CGSize) -> CGSize {
+
         if let size = self[constraint] {
+            return size
+        } else {
+            let size = calculation(constraint)
+
+            /// 1) Cache the measured size for the given constraint.
+            self[constraint] = size
+
+            /// 2) Optimization: Cache the measured size, so if we ask for this
+            /// exact size later on, we don't re-measure the element.
+            ///
+            /// This avoids a cache miss later on when a layout lays out an item with a measured size.
+            self[SizeConstraint(size)] = size
```

Comment:
Yeahhhhh, I was thinking about this too. It felt rare / weird enough that the tradeoff was worth it, but your point is valid – we can't guarantee folks _don't_ rely on this, so I am also fine just removing it... It _does_ cut the measurement cost in half once again though... Trying to think about how we could keep the "good" parts here...

Reply:
(I was also trying to figure out how to lift this level of caching up to this type only, but not really a doable thing rn..)

Reply:
Oh I have an idea

Reply:
Hmm, thinking about this more... This is OK? Eg you don't have to respect the size returned here. And if you still stretch the returned size, the aforementioned layout will still work... I think?

(Ok we've returned to where I was when I first wrote this code and had the same thought. Trying to figure out how to prove this case beyond "the snapshots in SPOS still pass".)

Reply:
After sleeping on it a bit, I guess I'm OK with this.

I do think that lifting this logic up to to be done explicitly in stack layout is still probably the more "right" thing, for a couple reasons:
1. the previously mentioned correctness argument, since you could definitely build an element that this breaks
2. the way that this interacts with stack layout is not very obvious. If someone else tweaked stack layout a bit we could lose this optimization without knowing why. Or tests would fail for frustratingly unclear reasons.

But practically speaking, I can only come up with hypothetical cases that cause (1), and I'm less worried about (2) knowing that a layout rewrite is likely right around the corner to support SPL. Worth the tradeoffs.

Reply:
I'm gonna hold off on merging now, and I would like to talk through this a bit more next week – I think these are very valid points and wanna make sure I don't paint us into a corner with this change. I also had a couple other ideas for how to lift / manage the caching a bit better for _only_ the stack layout which I need to play around with.

Reply:
I think I have a way to do this on the _get_ end, not the set end. Testing to see what happens.

Reply:
Actually wait, that won't work unless we have the concept of cached measurements and if they're natural sizes or not. I might just remove this for now, gonna play with it a bit more.

Reply:
Yeah, gonna just get rid of this for now. It reduces the benefit a bit, but it still cuts many measurements down to 2 passes instead of 2 * depth.

Reply:
Eg, with the flag disabled: 
<img width="991" alt="image" src="https://user-images.githubusercontent.com/327847/196571057-20878281-8a2e-40ba-bfd5-fd05345dd840.png">